### PR TITLE
Added Authentication

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

**Review by OleksiiTromsa — APPROVED** [review]
Very good job!
P.S. But "Added solution" is a bad naming for commit)

**Comment by Nokafan** [comment]
:) I promice to use some more efficient naming in the future.

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -92,20 +91,6 @@ public static void main(String[] args) {
         User userOne = authenticationService.register("userOne@ukr.net", "1234");
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/Main.java]: let's remove main method and class))

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 2]
```diff
@@ -15,6 +17,7 @@
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
+@Validated
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieController.java:20]: Why do you use this annotation?

File: src/main/java/com/dev/cinema/controllers/UserController.java [hunk 3]
```diff
@@ -4,25 +4,28 @@
...
     public UserController(UserService userService, UserMapper userMapper) {
         this.userService = userService;
-        this.userMapper = userMapper;
+        this.mapper = userMapper;
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto getByEmail(@RequestParam String email) {
-        return userMapper.fromUserToUserResponseDto(userService.findByEmail(email).orElseThrow());
+    public UserResponseDto getByEmail(Authentication authentication) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/UserController.java]: should we change this method? Let's leave it as it was

File: src/main/java/com/dev/cinema/dto/ticket/TicketRequestDto.java [hunk 4]
```diff
@@ -1,9 +1,12 @@
 package com.dev.cinema.dto.ticket;
 
+import javax.validation.constraints.Positive;
 import lombok.Data;
 
 @Data
 public class TicketRequestDto {
+    @Positive(message = "Movie session id should be positive")
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dto/ticket/TicketRequestDto.java]: коли ми використовуємо анотацію Positive чи можуть нам приходити елементи null?

File: src/main/java/com/dev/cinema/dto/user/UserRequestDto.java [hunk 5]
```diff
@@ -1,9 +1,11 @@
 package com.dev.cinema.dto.user;
 
+import javax.validation.constraints.Email;
 import lombok.Data;
 
 @Data
 public class UserRequestDto {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dto/user/UserRequestDto.java]: do we use this dto anywhere?

Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/user/UserRequestDto.java]: Not at the moment, but in the future will do probably, so I'll remove it.

File: src/main/java/com/dev/cinema/dto/ticket/TicketRequestDto.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/ticket/TicketRequestDto.java]: Блін про null не дочитав до кінця опис, обрадувався що є підходяща для чисел.

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/controllers/MovieController.java:20]: I was hoping this annotation works as general for the whole class, but iI am not sure, is it the correct translation. The description says that is for group validation.

File: src/main/java/com/dev/cinema/Main.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/Main.java]: Done.

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 2 above)
Comment (Sofasmile) [reply src/main/java/com/dev/cinema/controllers/MovieController.java:20]: you can see an example here
https://www.baeldung.com/spring-valid-vs-validated
it doesn't work as a general for whole class

**Review by Sofasmile — APPROVED** [review]

File: src/main/java/com/dev/cinema/controllers/UserController.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/controllers/UserController.java]: Fixed.

**Pull request merged**